examples/scripts/update_powershell.ps1

```diff
@@ -0,0 +1,386 @@
+<#
+.SYNOPSIS
+    Used to upgrade powershell to the target version with the required
+    pre-requisites list .NET Framework 4.5.2.
+.DESCRIPTION
+    The script will upgrade the powershell version to whatever is supplied as
+    the 'target_version' on the host (see PARAMETER target_version for
+    supported targets).
+
+    This script can be run on the following OS'
+        Windows Server 2008 (with SP2) - only supported target_version 3.0
+        Windows Server 2008 R2 (with SP1)
+        Windows Server 2012
+        Windows Server 2012 R2
+        Windows Server 2016
+
+        Windows 7 (with SP1)
+        Windows 8.1
+        Windows 10
+    
+    All OS' can be upgraded to 5.1 except for Windows Server 2008. If running
+    on Powershell 1.0 then this script will first upgrade the version to 2.0
+    before running the checks. This is because a lot of the upgrade paths need
+    this version installed as a baseline. If the .NET Framework version
+    installed is less than 4.5.2, it will be upgraded to 4.5.2 as this is
+    supported on all hosts and is required for v5.0.
+
+    As multiple packages can be installed in this process, multiple reboots may
+    be required to continue with the install. If a reboot is required the
+    script will detech if the 'username' and 'password' parameters have been
+    supplied. If they have been supplied it will automatically reboot and login
+    to continue the install process until it is all complete. If these
+    parameters are not set then it will prompt the user for a reboot and
+    require the user to log back in manually after the reboot before
+    continuing.
+
+    A log of this process is created in
+    $env:SystemDrive\temp\update_powershell.log which is usually C:\temp\. This
+    log can used to see how the script faired after an automatic reboot.
+.PARAMETER target_version
```

Comment:
Looks perfect, the only thing I was concerned with is using underscores in option names. I don't like them :-)

Reply:
I can probably just change it to `version`, that way people can write `-Version` or `-version` to their heart's content.